indico/modules/announcement/templates/display.html

```diff
@@ -1,23 +1,80 @@
-<div class="announcement-bar weak-hidden" id="announcement-bar">
-    <div class="flexrow f-self-stretch f-j-center f-a-center">
-        <i class="icon icon-warning"></i>
-        <div>{{- message | safe -}}</div>
-    </div>
-    <span class="icon-close close-button"></span>
-</div>
-<script>
-    (function() {
-        'use strict';
+{% macro render_announcement_legacy(type, message, persistent=false) %}
+    {% if type == 'html' %}
+        {{ message | safe }}
+    {% else %}
+        <div class="announcement-bar weak-hidden {{type}}" data-hash={{message | crc32 | string | tojson}}>
```

Comment:
`data-hash="{{message | crc32 | string | tojson }}"`

Reply:
Maybe I'm missing something, but shouldn't `tojson` already add the quotes?

Reply:
yeah, but using a json encoder to generate the outer quotes for an html attribute... that's blacker than black magic

Reply:
Agreed.